week02/Add_Two_Numbers.cpp

```diff
@@ -0,0 +1,29 @@
+// TC: O(N)
+// MC: O(1)
+
+class Solution {
+public:
+    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
+        ListNode* dummyHead = new ListNode();
+        ListNode* current = dummyHead;
+        int carry = 0;
+
+        while (l1 || l2) {
+            int x = l1 != nullptr ? l1 -> val : 0;
+            int y = l2 != nullptr ? l2 -> val : 0;
+
+            int sum = x + y + carry;
+            current -> next = new ListNode(sum % 10);
```

Comment:
As you're creating new nodes with `new`, the space complexity can never be constant. :)

Reply:
Thanks for pointing it out vai. I missed the memory here. I am creating N nodes inside the loop, so it is O(N) MC.